Update Documentations and Environment

agpenton opened the pull request:
<Include a few sentences describing the overall goals for this Pull Request. Please do not remove the checklist below, pull requests missing it will be ignored.>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate it if you help us to do so by completing the following items. Feel free to ask in a comment if you have trouble with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**


For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

* webUI step added.
* Documentation updated for the new Environment.
* docker-compose.override.yml for interactive test.

agpenton committed:
Changes in the repository:
- delete of the logs in the cite test Makefile.
- README.md modified for the new steps.
- Documentation added with the manual and automated tests.

agpenton committed:
upstream sync

agpenton committed:
CITE Test Guide updated

agpenton committed:
Update Documentations and Environment:
- webUI step added.
- Documentation updated for the new Environment.
- docker-compose.override.yml for interactive test.

agpenton committed:
shortening syntax in Makefile

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
-.. important::
+       .. code:: shell
+
+          git clone https://github.com/randomorder/geoserver.git --branch GSIP-176
```
The guide should not refer to a particular branch and developer repository, the PR will be eventually merged and people will use the GeoServer official one:

```suggestion
          git clone https://github.com/geoserver/geoserver.git
```

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+**Running the suite tests.**
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+   There is 2 way to run the suites, one is running with make that will
```
```suggestion
   There is 2 way to run the suites, one is running with `make` that will
```

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+
+         .. code:: shell
+
+            make clean suite=<suite-name>
```
Confused, so suite is an environment variable (you said so above), a make parameter, or both? And what about the war url, is it usable as a parameter in the call to make as well?

agpenton replied:
the suite is an environment variable and the war-url is used inside the `Makefile` to build the GeoServer.

randomorder replied:
"the suite is an environment variable" 

make it explicit in the doc then

"the war-url is used inside the Makefile to build the GeoServer."

can you make it a parameter of the makefile. How long would it take?

agpenton replied:
> "the suite is an environment variable"
> 
> make it explicit in the doc then
> 
done.

> "the war-url is used inside the Makefile to build the GeoServer."
> 
> can you make it a parameter of the makefile. How long would it take?

if you want to change the war_url to another URL you can do it on the command...since is a variable in the `makefile`, just that the value declared there is the default value. ex.

``` shell
make build suite=wcs10 war_url="https://build.geoserver.org/geoserver/main/geoserver-main-latest-war.zip"
```

aaime replied:
Then avoid talking about the env variables before, and just whow the command line parameters. You make people specify the same information twice (see my comment above, where the env variables are declared above), it's confusing.

agpenton replied:
solved

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+
+            make test suite=<suite-name>
+
+      -  And the last, but no less important run the full automate
```
```suggestion
      -  All of the above can also be run with a single command, using:
```

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+
+         .. note::
+
+            The first Docker build may take a long time.
```
This should be specified in the build command above (I've played commands one by one as instructed, and wondered why you're telling me it might be long just now).

agpenton replied:
fixed in this commit https://github.com/randomorder/geoserver/pull/3/commits/3f79a0684747816e281346ea4353d28e1930f565

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+
+   2. Running the test in the WebUI.
+
+      - To run the test in the WebUI, you should change the ``command`` parameter in the ``docker-compose.override.yml``.
```
Having to edit a file is not going to be acceptable.

That said... I see instructions below on how to run tests locally. So wondering what's the use of this section.

If it's needed, then read below.

The developer that's working on fixing GeoServer will have this file marked as modified by git, and then will do changes to the GeoServer own code to make it pass the suite.
A distraction, and they will commit the change to the docker setup along with the code fixes to GeoServer.
Please make it so that running the web UI is a configuration that does not need to edit files (e..g, a command line parameter to the makefiles)

agpenton replied:
this was solved.

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+
+How to run TEAM Engine standalone
+---------------------------------
+- To run a standalone version of TEAM Engine, start it with the following command:
```
What is the standalone version? What test suites does it contain?

agpenton replied:
i got the message, i deleted that part of the guide, resolved in https://github.com/randomorder/geoserver/pull/3/commits/44c57d046e9b90e86850565b4a51f6d1b203ea50

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+   #. ``Enable tests with multiple namespaces`` tests included
 
-     createdb -T template_postgis -U cite cite
+      .. image:: ./image/tewfs-1_0.png
```
Got a bunch of warnings from Sphinx (which cause a build error in GeoServer, we don't accept any of those):

```

sphinx:
     [echo] Running sphinx-build -D release=2.20-SNAPSHOT -W -b html -d "/home/aaime/devel/git-gs/doc/en/target/developer/doctrees" . "/home/aaime/devel/git-gs/doc/en/target/developer/html"
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:9: WARNING: Explicit markup ends without a blank line; unexpected unindent.
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:204: WARNING: Title level inconsistent:
     [exec] 
     [exec] Requirements:
     [exec] ~~~~~~~~~~~~~
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:197: WARNING: Duplicate explicit target name: "geoserver repository".
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewfs-1_0.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewfs-1_1.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewms-1_1a.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewms-1_1b.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewcs-1_0a.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewcs-1_0b.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewcs-1_0c.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:: WARNING: image file not readable: cite-test-guide/./image/tewcs-1_1a.png
     [exec] /home/aaime/devel/git-gs/doc/en/developer/source/cite-test-guide/index.rst:23: WARNING: undefined label: teamengine web application (if the link has no caption the label must precede a section header)

BUILD FAILED
/home/aaime/devel/git-gs/doc/en/build.xml:90: The following error occurred while executing this line:
/home/aaime/devel/git-gs/doc/en/build.xml:101: exec returned: 1
```

I see the images in the commit, believe you placed the images in the wrong location.

aaime replied:
To run a build of just the developer docs, have "ant" ready, go to your GeoServer checkout, and use:

```
cd doc/en
ant developer
```

agpenton replied:
the documentation doesn't have any error when get build, I guess we can resolve this conversation

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+- PostgreSQL with PostGIS extension installed. (only for the WFS Test Suite)
 
-   .. code:: SQL
+- Teamengine Running in docker container.
```
What does this mean though? Running "stand alone"? Or with the process that requires changing the compose files?

agpenton replied:
i got the message, i deleted that part of the guide

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
+#. Go to the browser and open the URL: http://localhost:8888/teamengine/
 
-     ant wfs-1.1
+   - after the site open, you need to create an account to run the tests.
```
Can you please bake into the docker image a fixed account (like ``cite/cite``)? It's annoying to have to create an account every time (our old CITE infrastructure had this)

agpenton replied:
this topic is resolved in this commit
https://github.com/randomorder/geoserver/pull/3/commits/3f79a0684747816e281346ea4353d28e1930f565

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -17,120 +18,294 @@ Check out CITE suite tools
    With the following parameters:
 
-   #. ``Capabilities URL`` http://localhost:8080/geoserver/wfs?service=wfs&request=getcapabilities&version=1.1.0
+   #. ``Capabilities URL`` http://<ip.of.the.goserver>:8080/geoserver/wfs?service=wfs&request=getcapabilities&version=1.1.0
```
This is not enough, the GeoServer must be using a given data dir and setup the database with the right test data.

For example, see this one:
https://github.com/geoserver/geoserver/tree/main/data/citewfs-1.0

It had instructions on how to run the setup.  Please add this information for each test suite, so that the dev charged to fix GeoServer to pass a given suite, knows how to setup their computer.

agpenton replied:
this topic was resolved in this commit https://github.com/randomorder/geoserver/commit/3f79a0684747816e281346ea4353d28e1930f565

randomorder committed:
Update doc/en/developer/source/cite-test-guide/index.rst

Co-authored-by: Andrea Aime <andrea.aime@gmail.com>

randomorder committed:
Update doc/en/developer/source/cite-test-guide/index.rst

Co-authored-by: Andrea Aime <andrea.aime@gmail.com>

randomorder committed:
Update doc/en/developer/source/cite-test-guide/index.rst

Co-authored-by: Andrea Aime <andrea.aime@gmail.com>

agpenton committed:
Changes include in this commit:
- user/pass for the teamengine, to avoid the creation everytime.
- Added more documetation.
- route of the images fixed.

agpenton committed:
cleaning the Makefile

agpenton committed:
merging

agpenton committed:
fixing some errata

randomorder committed:
Update index.rst

update description in the guide

randomorder committed:
Update index.rst

update war URL (new branch name)

agpenton committed:
typo in the guide

agpenton committed:
Merge branch 'GSIP-176-update' of https://github.com/agpenton/geoserver into GSIP-176-update

agpenton committed:
typo in the guide

randomorder committed:
Update index.rst

Update teamengine docker repo to use OSGeo Nexus https://repo.osgeo.org/#browse/browse:docker:v2%2Fgeoserver-cite%2Ftags%2Fteamengine_latest

agpenton committed:
typo in the guide

agpenton committed:
Merge branch 'GSIP-176-update' of https://github.com/agpenton/geoserver into GSIP-176-update

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,389 @@ Cite Test Guide
+
+       .. code:: shell
+
+          git clone https://github.com/geosolutions-it/geoserver.git
```
Remember to switch it to geoserver/geoserver before this one gets proposed in the GeoServer official PR. (think I already commented once about this)

agpenton replied:
fixed in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

agpenton replied:
Solved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,389 @@ Cite Test Guide
+            build: $(suite)         Will Build the GeoServer Docker Image for the Environment.
+            test: $(suite)      Will running the Suite test with teamengine.
+
+      - Choose which test to run by setting the Suite environment variable:
```
This is the bit that looks confusing, it seems that one has to go to the command line and export this environemtn variable, but following the instructions later, it's passed as a parameter to the make commands.

agpenton replied:
changed in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,389 @@ Cite Test Guide
+
+         .. code:: shell
+
+            make clean build test suite=<suite-name>
```
Just tried to run it, got this:

```
make clean build test suite=wcs10
Cleaning up previous runs
docker-compose -f docker-compose.yml -f ./wcs10/docker-compose.override.yml \
      down --remove-orphans --rmi all -v
Stopping cite_geoserver_1 ... done
Removing cite_teamengine_1 ... done
Removing cite_geoserver_1  ... done
Removing network cite_default
Removing volume cite_gs_datadir
Removing volume cite_gs_logs
Removing volume cite_gs_gwc_cache_dir
Removing image ogccite/geoserver:wcs10
Removing image geoserver-docker.osgeo.org/geoserver-cite:teamengine_5.3
docker-compose -f docker-compose.yml -f ./wcs10/docker-compose.override.yml rm -vfs
No stopped containers
rm -Rf logs/
rm: impossibile rimuovere 'logs/s0001/error_log/log.txt': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e1039_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e690_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e973_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e639_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e648_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e706_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e1240_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e1051_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e1089_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e699_1/log.xml': Permesso negato
rm: impossibile rimuovere 'logs/s0001/d68e410_1/d68e677_1/log.xml': Permesso negato
```

Maybe my docker is mis-setup (tbh, it's the Linux Mint default install), but I'm confident others will stumble into this as well. Is there a better way to do it, or does one really have to use sudo to run these builds?

agpenton replied:
the problem is that teamengine is running with the root user so the logs are created by root.
that will be resolve in the next commit.

agpenton replied:
solved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime replied:
Sudo will likely ask for password right? Best to document make seemingly ends up asking for credentials, some might get pretty spooked by such request. @randomorder is there no way to map the root user of docker to a local user, so that the files are created by the user that started the docker images?

randomorder replied:
yes but is non trivial

typical workaround in these cases is you run the process in the docker container (that is defined at docker image build time) with a user that has the same UID of the user that is triggering the docker run (run time).

I'll double check available options with @agpenton

aaime replied:
If it cannot be worked around, it should be documented.

agpenton committed:
fixing some errata

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,389 @@ Cite Test Guide
+
+   .. code-block:: shell
+
+    cd <root of geoserver repository>
```
This gets me to the root of the checkout, no?

agpenton replied:
solved

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,389 @@ Cite Test Guide
+   .. code-block:: shell
+
+    cd <root of geoserver repository>
+    psql -U cite cite < cite_data_postgis2.sql
```
So this one fails because there is no "cite_data_postgis2.sql" file in the root of the checkout.
For it to work from the root of the geoserver repository, the path should be:

```
build/cite/wfs10/citewfs-1.0/cite_data_postgis2.sql
```

Please check all other paths and see if a similar issue is present, and fix as needed.

agpenton replied:
solved https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
-     cd <root of geoserver install>
-     export GEOSERVER_DATA_DIR=<root of geoserver sources>/build/cite/wfs11/citewfs-1.1
-     ./bin/startup.sh
+    cd <root of geoserver install>
```
Same problem as above I believe.

agpenton replied:
solved https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
-     export GEOSERVER_DATA_DIR=<root of geoserver sources>/build/cite/wfs11/citewfs-1.1
-     ./bin/startup.sh
+    cd <root of geoserver install>
+    export GEOSERVER_DATA_DIR=<root of geoserver sources>/data/citewfs-1.0
```
No, in this PR the data directories have been moved, the citewfs-1.0 one is at:
``/build/cite/wfs10/citewfs-1.0``

agpenton replied:
solved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
-     ./bin/startup.sh
+    cd <root of geoserver install>
+    export GEOSERVER_DATA_DIR=<root of geoserver sources>/data/citewfs-1.0
+    ./bin/startup.sh
```
Ah so you're assuming one has a GeoServer installation? Hmm.. no one of the core devs is going to do this, why run the binaries when you want to make changes to the sources? 
But some external user might want to run the CITE tests of a given GeoServer version, maybe with custom modifications or plugins, so it's not wrong either.

Suggestion, just tell people to
- set the GeoServer data directory to ``$GEOSERVER_GIT_CLONE/build/cite/wfs10/citewfs-1.0``
- start GeoServer

Ad then maybe make an example on how to do it using the binary packages, like above. The core devs will know better than following these instructions and will start GeoServer directly from their IDE instead.

agpenton replied:
solved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
 
-     ant wfs-1.0
-     
+     make webUI
```
If one follows you step by step, the command will not work. in which directory should one be in order for the make to work? Add a ``cd <the directory>`` command before it, with the right directory

agpenton replied:
solved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -139,22 +374,38 @@ Requirements:
+   - after the site open, click on the **Sign in** button and enter the user and password.
 
-   With the following parameters:     
+   .. note:: the Default username/password are **teamengine/teamengine**.
```
Tried, literal copy/paste of the username/password, did not work. What could it be? 

![image](https://user-images.githubusercontent.com/325433/114404831-5ce01d80-9ba6-11eb-990e-f6ec95e168a1.png)

agpenton replied:
the user/password works, I just tried.

aaime replied:
Yes, I expected that, that's why I asked "What could it be"?
Like, maybe I have an outdated build of the docker images?

Btw, tried a clean build of the cite tools and getting:

```
aaime@colossus ~/devel/git-gs/build/cite (cite_doc_update) $ make clean
Cleaning up previous runs
docker-compose -f docker-compose.yml -f .//docker-compose.override.yml \
      down --remove-orphans --rmi all -v
ERROR: .FileNotFoundError: [Errno 2] No such file or directory: '././/docker-compose.override.yml'
make: *** [Makefile:18: clean] Errore 1
```

agpenton replied:
that's why the $(suite) parameter in the `make clean $(suite)`, to clean the images created by the suite selected. 
ex:
suite=wfs11
geoserver: ogccite/geoserver:wfs11
postgresql: ogccite/postgres:wfs11
teamengine: geoserver-docker.osgeo.org/geoserver-cite:teamengine_5.3

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -139,22 +374,38 @@ Requirements:
-   With the following parameters:     
+   .. note:: the Default username/password are **teamengine/teamengine**.
+
+   With the following parameters:
```
One has to do "create new session" and choose the test, like:

![image](https://user-images.githubusercontent.com/325433/114405009-89943500-9ba6-11eb-9c38-b3e64615d1fd.png)

![image](https://user-images.githubusercontent.com/325433/114405063-96188d80-9ba6-11eb-8d95-9c578d5c0269.png)

Only after this the parameters below will be asked.

agpenton replied:
solved https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
+
+#. Go to the browser and open the URL: http://localhost:8888/teamengine/
+
+   - after the site open, click on the **Sign in** button and enter the user and password. 
```
Does not say which ones (it's written down below for other tests thought)

agpenton replied:
solved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
+#. Go to the browser and open the URL: http://localhost:8888/teamengine/
+
+   - after the site open, click on the **Sign in** button and enter the user and password. 
+
```
One has to do "create new session" and choose the test, like:

![image](https://user-images.githubusercontent.com/325433/114405009-89943500-9ba6-11eb-9c38-b3e64615d1fd.png)

![image](https://user-images.githubusercontent.com/325433/114405063-96188d80-9ba6-11eb-8d95-9c578d5c0269.png)

Only after this the parameters below will be asked.

agpenton replied:
resolved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
    With the following parameters:
 
-   #. ``Capabilities URL`` http://localhost:8080/geoserver/wfs?request=getcapabilities&service=wfs&version=1.0.0
+   #. ``Capabilities URL`` http://<ip.of.the.goserver>:8080/geoserver/wfs?request=getcapabilities&service=wfs&version=1.0.0
```
Rigth, so it really needs to be the IP, "localhost" won't work since it would get the localhsot of the docker image running the teamengine. Maybe adding a warning somewhere about it would be good (a paragraph of things to pay attention to, before the list what to do for each CITE test, would be good)

agpenton replied:
resolved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,366 @@ Cite Test Guide
+   #. ``Enable tests with multiple namespaces`` tests included
 
-      .. image:: tewfs-1_0.png
+      .. image:: ./image/tewfs-1_0.png
```
After running the test the report seems missing. It's not true, but the output can be confusing.

In that paragraph of things to pay attention to, before the list of suites, tell developers that they should expect for some old test suites an output like:

![image](https://user-images.githubusercontent.com/325433/114408032-50a98f80-9ba9-11eb-83cc-87e703a4669d.png)

The first sentence is confusing, seems like one has to drop on the comand line and look at XML, but instead one can follow the link to the "detailed old test report" and then the report is there and it's usable:

![image](https://user-images.githubusercontent.com/325433/114408209-7a62b680-9ba9-11eb-8b19-3f8c7c493bd6.png)

agpenton replied:
resolved in this commit https://github.com/randomorder/geoserver/pull/3/commits/36a49e75d9c99b3948a52a63f21cd60741c65e1b

aaime reviewed (changes requested):
@agpenton I did another run of review. What I noted on the WFS 1.0 cite test instructions is often valid for other test suites as well, I did not repeat the same comments on each one of them. Please make sure to fix them all though.

agpenton committed:
Changes in the Documentation, and the Makefile syntax.

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,410 @@ Cite Test Guide
-  .. code:: shell
+   - the default url for the teamengine webUI is http://localhost:8888/teamengine/
+
+   - The output of the old suite tests, i might not appear in the Result page. So you should click on the link below **detailed old test report**, to get the full report. Ex.
```
```suggestion
   - The output of the old suite tests might not appear in the Result page. So you should click on the link below **detailed old test report**, to get the full report. Ex.
```

agpenton replied:
solved in https://github.com/randomorder/geoserver/pull/3/commits/91da4cbdc686ba5397014ab7887381ee240f8a1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,410 @@ Cite Test Guide
+
+   - Add the ip of the geoserver to the docker-compose file, so later you can use that as a host for the url of the tests.
+
+      - Go to the directory ``interactive``.
```
This approach makes people modify file in the same repository as GeoServer, while they are working to change the GeoServer source code, to make it pass the CITE tests. We have been over this before, it should not be done, sooner or later someone will commit their specific machine IP and fail.
They can also be on DHCP, with unstable IP addresses.

Best to tell them to just look up their IP address, and enter it i the URLs instead of localhost (they just need to be told to do so, do so here, it's a good place.

agpenton replied:
solved in https://github.com/randomorder/geoserver/pull/3/commits/91da4cbdc686ba5397014ab7887381ee240f8a1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -6,131 +6,410 @@ Cite Test Guide
+
+   #. ``Capabilities URL`` http://geoserver:8080/geoserver/wfs?service=wfs&request=getcapabilities&version=1.1.0
 
    #. ``Supported Conformance Classes``:
```
Actually GeoServer supports also locking, so it should be:

![image](https://user-images.githubusercontent.com/325433/114538712-f580a700-9c53-11eb-9e61-d4e6a8287ef8.png)

agpenton replied:
solved in https://github.com/randomorder/geoserver/pull/3/commits/91da4cbdc686ba5397014ab7887381ee240f8a1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -139,22 +418,36 @@ Requirements:
 
-   With the following parameters:     
+    cd <root of geoserver install>
+    export GEOSERVER_DATA_DIR=<root of geoserver sources>/data/citewms-1.1
```
Still points to the old, non existing directory.

agpenton replied:
solved in https://github.com/randomorder/geoserver/pull/3/commits/91da4cbdc686ba5397014ab7887381ee240f8a1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -170,59 +463,41 @@ Run WMS 1.1 tests
 
-#. Accept the default values and click ``Submit``
+    cd <root of geoserver install>
+    export GEOSERVER_DATA_DIR=<root of geoserver sources>/data/citewcs-1.0
```
Still points to the old, non existing directory.

agpenton replied:
solved in https://github.com/randomorder/geoserver/pull/3/commits/91da4cbdc686ba5397014ab7887381ee240f8a1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -237,39 +512,59 @@ Run WCS 1.0 tests
+   .. code-block:: shell
+
+    cd <root of geoserver install>
+    export GEOSERVER_DATA_DIR=<root of geoserver sources>/data/citewcs-1.1
```
Still points to the old, non existing directory.

agpenton replied:
solved in https://github.com/randomorder/geoserver/pull/3/commits/91da4cbdc686ba5397014ab7887381ee240f8a1b

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -237,39 +512,59 @@ Run WCS 1.0 tests
    #. ``Schemas``:
 
-      * Ensure that ``original schemas`` is selected
+      * Ensure that ``The server implements the original schemas from the WCS 1.0.0 scpecification (OGC 03-065`` is selected
```
```suggestion
      * Ensure that ``The server implements the original schemas from the WCS 1.0.0 specification (OGC 03-065`` is selected
```

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -139,22 +418,36 @@ Requirements:
-   .. image:: tewfs-1_1.jpg
+   .. image:: ./image/tewfs-1_1.png
 
 Run WMS 1.1 tests
```
Good thing that we have WMS 1.1, but we're missing WMS 1.3, please add it.

agpenton replied:
solved with https://github.com/randomorder/geoserver/pull/3/commits/47a6f7ee076dfe52c579b644dffa7142d8e97d4c

agpenton committed:
fixing some errata

agpenton committed:
Update doc/en/developer/source/cite-test-guide/index.rst

typo in the documentation.

Co-authored-by: Andrea Aime <andrea.aime@gmail.com>

agpenton committed:
Adding CITE wms13 tests to guide

agpenton committed:
Merge branch 'GSIP-176-update' of https://github.com/agpenton/geoserver into GSIP-176-update

aaime commented on doc/en/developer/source/cite-test-guide/index.rst:
```diff
@@ -237,39 +495,103 @@ Run WCS 1.0 tests
+
+   #. ``Capabilities URL``:
+
+         http://<ip-of-the-geoserver>:8080/geoserver/wcs?service=wcs&request=getcapabilities&version=1.1.1
```
No, won't work if there is ``service=wcs`` in the link, cause the test suite adds a ``service=WCS`` in addition to it, and makes the request fail. 
Should just be:
```suggestion
         http://<ip-of-the-geoserver>:8080/geoserver/wcs
```
The test suite is adding (duplicating) all other parameters

agpenton committed:
Update doc/en/developer/source/cite-test-guide/index.rst

changing the URL of the wcs11 test

Co-authored-by: Andrea Aime <andrea.aime@gmail.com>
